cmake/SetupCompilers.cmake

```diff
@@ -9,16 +9,17 @@ set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3" CACHE STRING "")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -O3" CACHE STRING "")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0" CACHE STRING "")
 
-if (RAJA_ENABLE_MODULES AND CMAKE_CXX_COMPILER_ID MATCHES Clang)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fmodules")
+if (RAJA_ENABLE_MODULES)
+  message(WARNING "In the next RAJA Release, Modules will be deprecated and should not be used.")
+  set(RAJA_ENABLE_MODULES Off CACHE BOOL "" FORCE)
```

Comment:
Added this in the cmake to say if the RAJA_ENABLE_MODULES flag is used, print out a warning that it's deprecated and force it to be turned off instead - hope that's ok?

Reply:
Yes. I think you could add to the message that the flag `-fmodules` can be added manually if desired.

Reply:
Sounds good!